src/vaults/BalanceSheet.sol

```diff
@@ -293,7 +299,8 @@ contract BalanceSheet is Auth, Recoverable, IBalanceSheet, IBalanceSheetGatewayH
 
     function _executeRevoke(PoolId poolId, ShareClassId scId, address from, uint128 shares) internal {
         IShareToken token = poolManager.shareToken(poolId, scId);
-        token.burn(from, shares);
+        token.authTransferFrom(from, from, address(this), shares);
```

Comment:
Means a manager does not need to provide an allowance before calling `BalanceSheet.revoke`. Similar to how no allowance is required for `requestRedeem` and `transferShares`